python/pyiceberg/utils/config.py

```diff
@@ -0,0 +1,130 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import logging
+import os
+from typing import Any, List, Optional
+
+import yaml
+
+from pyiceberg.typedef import EMPTY_DICT, FrozenDict, RecursiveDict
+
+PYICEBERG = "pyiceberg__"
+CATALOG = "catalog"
+HOME = "HOME"
+PYICEBERG_HOME = "PYICEBERG_HOME"
+PYICEBERG_YML = ".pyiceberg.yml"
```

Comment:
I thought the extension for YAML was [`.yaml`](https://yaml.org/faq.html).

Reply:
According to Wiki it is both:
![image](https://user-images.githubusercontent.com/1134248/184511299-434265a3-f0de-4a18-bbf5-b39b1d7f5053.png)

But the yaml faq recommends `.yaml`: https://yaml.org/faq.html

Interestingly enough, `dbt` uses `.yml` for their profiles https://docs.getdbt.com/dbt-cli/configure-your-profile

Let me change it to `.yaml`